tests/test_generate_mbz_toc.py

```diff
@@ -0,0 +1,202 @@
+    mocker, tmp_path, page_builder, lesson_builder, mbz_builder
+):
+    activities = []
+    sections = []
+    page_html = "<div><p>Page</p></div>"
+    for indx in range(1):
+        activities.append(page_builder(
+            section_id=2,
+            id=indx,
+            name="4 Page",
+            html_content=page_html
+        ))
+
+    lesson_1_name = "1 Lesson Page"
+    lesson_2_name = "2 Lesson Page"
+    lesson_3_name = "3 Lesson Page"
+    lesson_1_html = f"<div><p>{lesson_1_name} content</p></div>"
+    lesson_2_html = f"<div><p>{lesson_2_name} content</p></div>"
+    lesson_3_html = f"<div><p>{lesson_3_name} content</p></div>"
+    for indx in range(2):
+        activities.append(lesson_builder(
+            section_id=indx,
+            id=indx,
+            name="Lesson",
+            pages=[
+                {
+                    "id": 1,
+                    "title": lesson_1_name,
+                    "html_content": lesson_1_html,
+                    "prevpageid": 0,
+                    "nextpageid": 2
+                },
+                {
+                    "id": 2,
+                    "title": lesson_2_name,
+                    "html_content": lesson_2_html,
+                    "prevpageid": 1,
+                    "nextpageid": 3,
+                },
+                {
+                    "id": 3,
+                    "title": lesson_3_name,
+                    "html_content": lesson_3_html,
+                    "prevpageid": 2,
+                    "nextpageid": 0,
+                }
+            ]
+        ))
+
+    sections.extend([{
+            "id": 1,
+            "title": "First Section"
+        },
+        {
+            "id": 2,
+            "title": "Second Section"
+        }])
+
+    mbz_path = tmp_path / "mbz"
+    html_path = tmp_path / "html"
+    html_path.mkdir()
+
+    mbz_builder(mbz_path, activities=activities, sections=sections)
+    mbtools.extract_html_content.replace_content_tags(mbz_path, html_path)
+
+    md_file = "toc.md"
+    mocker.patch(
+        "sys.argv",
+        ["", f"{mbz_path}", str(tmp_path / md_file)]
+    )
+    main()
+    lines = []
+    with open((tmp_path / md_file), 'r') as fp:
+        lines = [line.rstrip() for line in fp]
+
+    ordering_list = []
+    entries = {}
+    for line in lines:
+        if (len(line.split("](")) > 1):
+            name = line.split("](")[0].strip('* [')
+            link = line.split("](")[1].split(')')[0]
+            entries[name] = link
+            ordering_list.append(name)
+
+    pre_list = ordering_list
+    ordering_list.sort()
+    assert pre_list == ordering_list
+
+    for name in entries.keys():
+        path = str(tmp_path) + entries[name].strip('.')
+        with open(path, 'r') as f:
+            if ('Lesson' in name):
+                content = f.read()
+                assert name in content
+            else:
+                content = f.read()
+                assert page_html in content
```

Comment:
In general, it's pretty hard for me to understand what this code is doing / checking for. I gave up trying to read it and instead tried to break it to suss out the behavior. However, it turns out I can completely break the generator by doing something like:

```python
def make_link(name, uuid):
    return "" 

def make_nested_bullet(level, content):
    return f"{' '*(level-1)*4}* \n"
```

so the generated MD is just a bunch of bullets...but the tests still pass. I would revisit these unit test implementations and perhaps:

1. Take a TDD approach to be sure they're actually failing if the implementation is broken
2. Try to see if we can write things in a way that's easier for someone to come along later and understand / modify.

Reply:
@rnathuji Alright, I completely scrapped what was there and re-implemented tests for this. Let me know what you think. Sorry for subjecting you to looking at that covid-induced jumble of nonsense 🫢, especially before the all hands presentation. I think I was very interested in "GETTING IT OVER WITH" on Wednesday 😅